docs/design.md

```diff
@@ -143,6 +143,12 @@ struct names.
   and the `typedef` name - i.e., `typedef bar { type enumeration { ... }}` in
   module `baz` is represented by an enumerated type named `Bar_Baz`.
   * This mapping is handled by `enumgen.go`:`resolveTypedefEnumeratedName`.
+* builtin `enumeration` types defined within a non-builtin (i.e. `typedef`)
```

Comment:
Can I suggest some alternate wording here?

```
* where an `enumeration` is defined within a `typedef` that contains a `union` - 
  the enumerated language type that is generated is named according to the name 
  of the `typedef` with `_Enum` appended to the name. 

  For example:

module bar {
  ...
  typedef baz {
     type union {
        type enumeration { ... }
        type string;
     }
  }
}

  would result in a type named `Bar_Baz_Enum` being generated in the output
  code.
```

Reply:
Thanks, this is much cleaner